cirq/__init__.py

```diff
@@ -393,7 +393,7 @@
     to_resolvers,
     to_sweep,
     to_sweeps,
-    TrialResult,
+    Result,
```

Comment:
Instead of removing `TrialResult`, let's keep it [marked](https://github.com/quantumlib/Cirq/blob/8e30cf6815dcb1ebbd0244243f49e64b7cd08366/cirq/_compat.py#L85) as deprecated (see [example](https://github.com/quantumlib/Cirq/blob/33194ad3a2f4c28c48417876ff7dd1a6732d3e41/cirq/linalg/states.py#L23)).

Reply:
Like this? Or do you just want me to mark it as deprecated for now? Happy either way, just trying to understand what you prefer.

Also, I realized, thanks to your comment, that I should have renamed a file which I just did.

Reply:
We should mark it as deprecated - i.e. anytime someone uses TrialResult, they should get a warning. The target deprecation release would be 0.11.0 for this.

Reply:
Moved to 0.11.0

Any suggestion on how to fix the errors? I am a bit confused, because it's complaining about the old object, which I intend to delete, right?